modules/metrics/src/main/java/com/spotify/apollo/metrics/semantic/SemanticRequestMetrics.java

```diff
@@ -61,60 +86,124 @@
+                  "what", REPLY_SIZE.what(),
+                  "unit", "B"
+              )));
+    } else {
+      replySizeHistogram = Optional.empty();
+    }
+
+    if (enabledMetrics.contains(REQUEST_SIZE)) {
+      requestSizeHistogram = Optional.of(
+          metricRegistry.histogram(
+              metricId.tagged(
+                  "what", REQUEST_SIZE.what(),
+                  "unit", "B"
+              )));
+    } else {
+      requestSizeHistogram = Optional.empty();
+    }
+
+    if (enabledMetrics.contains(ENDPOINT_REQUEST_DURATION)) {
+      timerContext = Optional.of(
+          metricRegistry
+              .timer(metricId.tagged("what", ENDPOINT_REQUEST_DURATION.what()))
+              .time());
+    } else {
+      timerContext = Optional.empty();
+    }
+
+    if (enabledMetrics.contains(DROPPED_REQUEST_RATE)) {
+      droppedRequests = Optional.of(
+          metricRegistry.meter(
+              metricId.tagged(
+                  "what", DROPPED_REQUEST_RATE.what(),
+                  "unit", "request"
+              )));
+    } else {
+      droppedRequests = Optional.empty();
+    }
 
     this.sentReplies = requireNonNull(sentReplies);
     this.sentErrors = requireNonNull(sentErrors);
 
-    registerRatioGauge(metricId, "1m", () -> RatioGauge.Ratio.of(this.sentErrors.getOneMinuteRate(),
-                                                                 this.sentReplies.getOneMinuteRate()));
-    registerRatioGauge(metricId, "5m", () -> RatioGauge.Ratio.of(this.sentErrors.getFiveMinuteRate(),
-                                                                 this.sentReplies.getFiveMinuteRate()));
-    registerRatioGauge(metricId, "15m", () -> RatioGauge.Ratio.of(this.sentErrors.getFifteenMinuteRate(),
-                                                                  this.sentReplies.getFifteenMinuteRate()));
+    registerRatioGauge(metricId, "1m", () -> Ratio.of(this.sentErrors.getOneMinuteRate(),
+                                                      this.sentReplies.getOneMinuteRate()));
+    registerRatioGauge(metricId, "5m", () -> Ratio.of(this.sentErrors.getFiveMinuteRate(),
+                                                      this.sentReplies.getFiveMinuteRate()));
+    registerRatioGauge(metricId, "15m", () -> Ratio.of(this.sentErrors.getFifteenMinuteRate(),
+                                                       this.sentReplies.getFifteenMinuteRate()));
   }
 
   private void registerRatioGauge(MetricId metricId,
                                   String stat,
-                                  Supplier<RatioGauge.Ratio> ratioSupplier) {
-    metricRegistry.register(
-        metricId.tagged("what", "error-ratio", "stat", stat),
-        new RatioGauge() {
-          @Override
-          protected Ratio getRatio() {
-            return ratioSupplier.get();
-          }
-        });
+                                  Supplier<Ratio> ratioSupplier) {
+    if (enabledMetrics.contains(ERROR_RATIO)) {
```

Comment:
Keep Consumer as member instead of SemanticMetricRegistry and perform check once in constructor